### PR TITLE
add health check

### PR DESCRIPTION
**Pull request by Nottt**
It tests both http and https, and if both fail exit with code 1.

### CODE REVIEW

**Comment by TheMeanCanEHdian** [comment]
What is the logic behind the 3s timeout rather than the 30s default? Additionally, this should also have --start-period set, probably 60s to play it safe when the container is starting up.

Edit: Let's say 90s for --start-period to account for a first-time startup when it clones the repro

**Comment by TheMeanCanEHdian** [comment]
Also, can you add -Ss to curl and redirect the request to /dev/null so it only outputs on errors?

File: Dockerfile [hunk 1]
```diff
@@ -76,3 +76,4 @@ ENTRYPOINT ["/init"]
 # ports and volumes
 VOLUME /config /plex_logs
 EXPOSE 8181
+HEALTHCHECK  --timeout=3s CMD curl -ILf http://localhost:8181 || curl -ILfk https://localhost:8181 || exit 1
```
Comment (Arcanemagus) [inline Dockerfile]: Like this 😉
```suggestion
HEALTHCHECK  --start-period=90s CMD curl -ILfSs http://localhost:8181 > /dev/null || curl -ILfkSs https://localhost:8181 > /dev/null || exit 1
```

**Comment by Nottt** [comment]
There is no logic in 3s timeout, it's just a value that lots of people use it, but it can be any arbitrary value. 

About the `--start-period` I believe it's not necessary. If you start the container with the healthcheck and without a timezone just to crash it, you'll see it already takes more than 60 seconds to start assessing the health: `Up About a minute (health: starting)`

Also what do you mean output nothing? There's already no output with `docker logs`... only with` docker inspect`:

```
"Health": {
                "Status": "healthy",
                "FailingStreak": 0,
                "Log": [
                    {
                        "Start": "2019-04-22T17:06:31.955889083Z",
                        "End": "2019-04-22T17:06:32.276000198Z",
                        "ExitCode": 0,
                        "Output": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\ncurl: (22) The requested URL returned error: 400 Bad Request\n  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0HTTP/1.1 303 See Other\r\n\r  0   100    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\n\r  0 27493    0     0    0     0      0      0 --:--:-- --:--:Content-Length: 100\r\nVary: Accept-Encoding\r\nServer: CherryPy/5.1.0\r\nLocation: https://localhost:8181/home\r\nDate: Mon, 22 Apr 2019 17:06:32 GMT\r\nContent-Type: text/html;charset=utf-8\r\n\r\nHTTP/1.1 200 OK\r\nDate: Mon, 22 Apr 2019 17:06:32 GMT\r\nVary: Accept-Encoding\r\nContent-Length: 27493\r\nContent-Type: text/html;charset=utf-8\r\nServer: CherryPy/5.1.0\r\n\r\n-- --:--:--     0\n"
```

Do you want to hide it from docker inspect too? Usually people don't do that...either way it's up to you guys

**Comment by Arcanemagus** [comment]
That's not what those commands do... `-Ss` combined with redirecting `stdout` to `/dev/null` means that only errors would end up in the results of `docker inspect`, as it should be.

The documentation at least says that the default value for the startup delay is `0s`, if it's showing `starting` with the default value maybe they have a different default from the documentation?

**Comment by Nottt** [comment]
Yeah I noticed this difference between their documentation and the practice... and oh, makes sense the curl thing now.

So what we do? Want me to commit your suggestion? I think 30s timeout is still a little high, but ok

**Comment by TheMeanCanEHdian** [comment]
Ya go ahead and commit the suggestions made. I'd rather err on the side of too high and adjust later if needed.

**Comment by Arcanemagus** [comment]
Thanks! 🎉
image

**Comment by ehvanr** [comment]
FYI this health check breaks if you have a custom port or http root in config.ini. Was driving me a little nuts wondering why my docker swarm network & reverse proxy wasnt working. 

For those who want a fix now, you can override in `docker-compose.yml`:

```
        healthcheck:
            test: ["CMD", "curl", "-ILfSs", "http://localhost:<http_port>/<http_root>"]
```

**Comment by Arcanemagus** [comment]
We have the ability to change the Tautulli port disabled in the settings inside the Docker container for _many_ reasons, this is one of them. You should be using Docker itself to remap the port instead of changing it within the container.

**Comment by michaelarnauts** [comment]
See above issue. This broke setups with basic auth enabled.

**Pull request merged**